test/unit/settings_test.rb

```diff
@@ -169,6 +179,26 @@ def test_require_cc_on_signup_visible_ui_switch_on_rolling_updates
     assert settings.monthly_billing_enabled
   end
 
+  test 'empty values sanitized for non-null columns' do
+    settings.update(public_search: true)
+    assert settings.reload.public_search
+
+    settings.update(public_search: "")
+    assert settings.reload.public_search
+
+    settings.update(public_search: nil)
```

Comment:
Is `nil` also converted to `true`? That's very weird.

Reply:
I think it doesn't result in `true` but is skipped from being updated.

Reply:
Maybe we should test that `false` works though and is not skipped.

Reply:
Yeah, this is not about `nil` being converted to `true`, but about filtering it out, because it causes `Mysql2::Error: Column 'public_search' cannot be null`

It's asserted that the value is true because this is the original value.

Reply:
Understood 👍